app/models/template_input.rb

```diff
@@ -91,11 +91,12 @@ def resolved_value
 
   class UserInputResolver < InputResolver
     def ready?
-      !input_value.nil?
+      @renderer.invocation
     end
 
     def resolved_value
-      input_value.value
+      raise(ValueNotReady, _("Required input not specified")) if @input.required? && input_value.try(:value).blank?
```

Comment:
so when I don't mark it as required and it has default value, what will be result of that input macro?